test/Microsoft.AspNetCore.Mvc.FunctionalTests/InputFormatterTests.cs

```diff
@@ -80,6 +80,24 @@ public InputFormatterTests(MvcTestFixture<FormatterWebSite.Startup> fixture)
         }
 
         [Theory]
+        [InlineData("application/json", "")]
+        [InlineData("application/json", "    ")]
+        public async Task JsonInputFormatter_IsModelStateValid_FalseForEmptyRequestBody(
```

Comment:
Good test but the name is misleading (yes, that's a problem in this class) Suggest `JsonInputFormatter_ReturnsBadRequest_ForEmptyRequestBody`.